docs/accessing-services.asciidoc

```diff
@@ -0,0 +1,168 @@
+
+[id="{p}-kubernetes-service"]
+=== Managing Kubernetes services
+
+For each resource, `Elasticsearch` or `Kibana`, the operator manages a Kubernetes service named `<name>-<kind>`, which is of type `ClusterIP` by default. `ClusterIP` exposes the service on a cluster-internal IP and makes the service only reachable from the cluster.
+
+[source,sh]
+----
+> kubectl get svc
+
+NAME                TYPE           CLUSTER-IP      EXTERNAL-IP      PORT(S)          AGE
+hulk-es             ClusterIP      10.19.252.160   <none>           9200:32361/TCP   25m
+hulk-kibana         ClusterIP      10.19.247.151   <none>           5601:32567/TCP   25m
+----
+
+[id="{p}-allow-public-access"]
+=== Allowing public access
+
+You can expose services in link:https://kubernetes.io/docs/concepts/services-networking/service/#publishing-services-service-types[different ways] by specifying an `http.service.spec.type` in the `spec` of a resource manifest.
+On cloud providers which support external load balancers, you can set the `type` field to `LoadBalancer` to provision a load balancer for the `Service`, and populate the column `EXTERNAL-IP` after a short delay. Depending on the cloud provider, it may incur charges.
+
+[source,yaml]
+----
+apiVersion: <kind>.k8s.elastic.co/v1alpha1
+kind: <Kind>
+metadata:
+  name: hulk
+spec:
+  version: 7.1.0
+  http:
+    service:
+      spec:
+        type: LoadBalancer
+----
+
+[source,sh]
+----
+> kubectl get svc
+
+NAME                TYPE           CLUSTER-IP      EXTERNAL-IP      PORT(S)          AGE
+hulk-es             LoadBalancer   10.19.252.160   35.198.131.115   9200:32361/TCP   25m
+hulk-kibana         LoadBalancer   10.19.247.151   35.242.197.228   5601:32567/TCP   25m
+----
+
+[id="{p}-static-ip-custom-domain"]
+=== Reserving Static IP and custom domain
+
+To use a custom domain name, you can reserve a static IP and/or use an Ingress instead of a `LoadBalancer` `Service`. Whatever you use, your DNS must be added to the certificate SAN in the `spec` of your Elastic resources manifest:
+
+[source,yaml]
+----
+spec:
+  http:
+    service:
+      spec:
+        type: LoadBalancer
+    tls:
+      selfSignedCertificate:
+        subjectAltNames:
+        - dns: hulk.example.com
+----
+
+[id="{p}-request-elasticsearch-endpoint"]
+== Requesting the Elasticsearch endpoint
+
+To request the Elasticsearch endpoint using a `LoadBalancer` `Service`, basic authentication and certificate validation with the CA,
+you have to retrieve the CA certificate, the IP of the service and the password of the elastic user:
+
+[source,sh]
+----
+NAME=hulk
+
+kubectl get secret "$NAME-ca" -o go-template='{{index .data "ca.pem" | base64decode }}' > ca.pem
+IP=$(kubectl get svc "$NAME-es" -o jsonpath='{.status.loadBalancer.ingress[].ip}')
+PW=$(kubectl get secret "$NAME-elastic-user" -o go-template='{{.data.elastic | base64decode }}')
+
+curl --cacert ca.pem -u elastic:$PW https://$IP:9200/
+----
```

Comment:
a) Should this document also include a section how to access Elasticsearch from inside the cluster? 
b) Otherwise I would rename the document to "How to access Elastic Stack services from outside of k8s" or similar

I think a) is preferable

Reply:
Your are right. We forgot to talk how it works from inside the cluster.
We can split 'Requesting the Elasticsearch endpoint' in 2 sections?

```
[id="{p}-request-elasticsearch-endpoint"]
== Requesting the Elasticsearch endpoint

[id="{p}-from-within-kubernetes-cluster"]
=== From within the Kubernetes cluster

To request the Elasticsearch endpoint from within the Kubernetes cluster,
you have to retrieve the CA certificate and the password of the `elastic` user.

[source,sh]
----
NAME=hulk

kubectl get secret "$NAME-ca" -o go-template='{{index .data "ca.pem" | base64decode }}' > ca.pem
PW=$(kubectl get secret "$NAME-elastic-user" -o go-template='{{.data.elastic | base64decode }}')

curl --cacert ca.pem -u elastic:$PW https://$NAME-es:9200/
----

[id="{p}-from-outside-kubernetes-cluster"]
=== From outside the Kubernetes cluster

To request the Elasticsearch endpoint from outside the Kubernetes cluster,
you have to retrieve the CA certificate, the password of the `elastic` user and the IP of the `LoadBalancer` `Service`.

[source,sh]
----
NAME=hulk

kubectl get secret "$NAME-ca" -o go-template='{{index .data "ca.pem" | base64decode }}' > ca.pem
IP=$(kubectl get svc "$NAME-es" -o jsonpath='{.status.loadBalancer.ingress[].ip}')
PW=$(kubectl get secret "$NAME-elastic-user" -o go-template='{{.data.elastic | base64decode }}')

curl --cacert ca.pem -u elastic:$PW https://$IP:9200/
----

At this stage, you should get this message:
...
etc
...
```